### ISSUE TITLE
Predefined analyzers require redundant name

### ISSUE THREAD

**Issue by akx** [elastic/elasticsearch-dsl-py#453]
Hey,

```python
m.field('foo', String(analyzer=EnglishAnalyzer())
```

complains about the missing name for the analyzer. As far as I understand, it's a constant `english`.

```python
m.field('foo', String(analyzer=EnglishAnalyzer(name='asdfoo'))
```

doesn't end up passing the `asdfoo` analyzer into the mapping at all, so naturally I get

> elasticsearch.exceptions.RequestError: TransportError(400, 'MapperParsingException[mapping [mapping]]; nested: MapperParsingException[Analyzer [asdfoo] not found for field [foo]]; ')

later.

TL;DR: I think the predefined analyzers should work without passing the redundant constant name in the constructor.

Versions in use:

```
elasticsearch==1.9.0
elasticsearch-dsl==0.0.11
```

**Comment by HonzaKral** [comment]
Good point, thanks for the report.

You can currently also just use the name of the analyzer (`'english'`) for builtin analyzers but I agree that the explicit syntax should work too without having to give a dummy name. After all explicit is better than implicit.

**Comment by HonzaKral** [comment]
May I ask how you are creating your `EnglishAnalyzer`?

I am afraid that the issue is slightly different - analyzers are instances, not classes. So the proper way to create an `english` analyzer would be to either use the string directly (`'english'`), create one using the `analyzer` function (`analysis.analyzer('english')`) or instantiate `BuiltinAnalyzer` directly (`analysis.BuiltinAnalyzer('english')`).

This is still an issue since it's not really consistent with other parts of the library, but works fine in my tests unless I am missing something.

Sorry for the confusion!

**Comment by HonzaKral** [comment]
Sorry, seem to be having a slow day when I type faster than I think or read :(

This has been resolved in 2.0.0 and newer when analyzers work as described above and we removed the `EnglishAnalyzer` class precisely for these reasons.

My apologies again for the confusion, I will close the ticket since it's fixed in a released version but feel free to reopen if you find that the new version doesn't work properly.

Thanks!

**Comment by akx** [comment]
No probs.

I'm just stuck on ES 1.7, and I understand the newer client versions won't work with that?
